### PR TITLE
fix(container_node_pool): panic interface conversion on `linux_node_config.sysctls`

### PR DESCRIPTION
**Pull request by toVersus**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/15847

* Add type assertion to extract empty values from `linux_node_config.sysctls` field
* Clean up existing test case to only pass single `sysctls` parameter to be changed
* Add test case to create a nodepool with empty `linux_node_config.sysctls`

Without the fix, the test case to create a nodepool with empty `linux_node_config.sysctls` will fail.

```console
❯ make testacc TEST=./google/services/container TESTARGS='-run=TestAccContainerNodePool_withLinuxNodeConfig'
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/container -v -run=TestAccContainerNodePool_withLinuxNodeConfig -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccContainerNodePool_withLinuxNodeConfig
=== PAUSE TestAccContainerNodePool_withLinuxNodeConfig
=== CONT  TestAccContainerNodePool_withLinuxNodeConfig
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1276 [running]:
github.com/hashicorp/terraform-provider-google/google/services/container.expandLinuxNodeConfig(...)
	/Users/tsubasanagasawa/go/src/github.com/hashicorp/terraform-provider-google/google/services/container/node_config.go:927
github.com/hashicorp/terraform-provider-google/google/services/container.expandNodeConfig({0x1071ec840?, 0x14001a923d8?})
	/Users/tsubasanagasawa/go/src/github.com/hashicorp/terraform-provider-google/google/services/container/node_config.go:849 +0x1808
github.com/hashicorp/terraform-provider-google/google/services/container.expandNodePool(0x1079eb778?, {0x0, 0x0})
	/Users/tsubasanagasawa/go/src/github.com/hashicorp/terraform-provider-google/google/services/container/resource_container_node_pool.go:867 +0x2f8
github.com/hashicorp/terraform-provider-google/google/services/container.resourceContainerNodePoolCreate(0x140029c3380, {0x107989740?, 0x14000f32400?})
	/Users/tsubasanagasawa/go/src/github.com/hashicorp/terraform-provider-google/google/services/container/resource_container_node_pool.go:477 +0x120
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1079e2678?, {0x1079e2678?, 0x1400140e630?}, 0xd?, {0x107989740?, 0x14000f32400?})
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:695 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140007ad500, {0x1079e2678, 0x1400140e630}, 0x140019cbc70, 0x140029c3200, {0x107989740, 0x14000f32400})
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:837 +0x888
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000eab110, {0x1079e2678?, 0x1400140e570?}, 0x140024650e0)
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1021 +0xb78
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0x140010720c0, 0x14001072120, {0x1400230eee0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140024483c0, {0x1079e2678?, 0x140013d1d40?}, 0x1400196c770)
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x107904520?, 0x140024483c0}, {0x1079e2678, 0x140013d1d40}, 0x1400196c700, 0x0)
	/Users/tsubasanagasawa/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000c22000, {0x1079eb898, 0x140027c7380}, 0x140019bdb00, 0x14001b552c0, 0x108f5a8e0, 0x0)
	/Users/tsubasanagasawa/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:1360 +0xcb4
google.golang.org/grpc.(*Server).handleStream(0x14000c22000, {0x1079eb898, 0x140027c7380}, 0x140019bdb00, 0x0)
	/Users/tsubasanagasawa/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:1737 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/Users/tsubasanagasawa/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:982 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/tsubasanagasawa/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:980 +0x16c
FAIL	github.com/hashicorp/terraform-provider-google/google/services/container	331.318s
FAIL
```

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
container: fixed an issue in `google_container_node_pool` where empty `linux_node_config.sysctls` would crash the provider
```

### CODE REVIEW

**Comment by toVersus** [comment]
I removed the suspicious code that was affecting the VCR test for the remaining nodepool test cases. However, I'm not sure how to handle the fact that the following change is always planned when reapplying after creating a nodepool.

Terraform code:

```HCL
resource "google_container_cluster" "cluster" {
  name               = "test-cluster"
  location           = "us-central1-f"
  initial_node_count = 1
}

resource "google_container_node_pool" "np" {
  name               = "test-nodepool"
  location           = "us-central1-f"
  cluster            = google_container_cluster.cluster.name
  initial_node_count = 1
  node_config {
    image_type = "COS_CONTAINERD"
    linux_node_config {
      sysctls = {}
    }
    oauth_scopes = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
    ]
  }
}
```

Output:

```console
TF_LOG=DEBUG TF_LOG_PATH=plan.log TF_CLI_CONFIG_FILE="$HOME/tf-dev-override.tfrc" terraform apply

(...)

Terraform will perform the following actions:

  # google_container_node_pool.np will be updated in-place
  ~ resource "google_container_node_pool" "np" {
        id                          = "projects/nagasawa-test/locations/us-central1-f/clusters/test-cluster/nodePools/test-nodepool"
        name                        = "test-nodepool"
        # (9 unchanged attributes hidden)

      ~ node_config {
            tags              = []
            # (15 unchanged attributes hidden)

          + linux_node_config {}

            # (1 unchanged block hidden)
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

I think this warning is the route cause of this issue but how can I avoid that?

```console
2023-09-19T11:34:07.352+0900 [WARN]  Provider "registry.terraform.io/hashicorp/google" produced an invalid plan for google_container_node_pool.np, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      (...)
      - .node_config[0].linux_node_config[0].sysctls: planned value cty.NullVal(cty.Map(cty.String)) does not match config value cty.MapValEmpty(cty.String)
      (...)
```

File: mmv1/third_party/terraform/services/container/node_config.go.erb [hunk 1]
```diff
@@ -1008,7 +1008,10 @@ func expandLinuxNodeConfig(v interface{}) *container.LinuxNodeConfig {
 	if len(ls) == 0 {
 		return nil
 	}
-	cfg := ls[0].(map[string]interface{})
+	cfg, ok := ls[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/container/node_config.go.erb:1014]:
```suggestion
	if ls[0] == nil {
		return &container.LinuxNodeConfig{}
	}
	cfg := ls[0].(map[string]interface{})
```
Can we try something like this and see if the API will return back the value?

**Review by shuyama1 — COMMENTED** [review]
Thanks for making a fix for this issue. Yeah, explicitly added it in the flattener to write back `linux_node_config {}` that will potentially break all current configurations and we should avoid doing so. I wonder if we explicitly send the empty fields would the API return it back as the current issues is that the API does not return `linux_node_config {}` and Terraform will try to update it since it's in the config.

File: mmv1/third_party/terraform/services/container/node_config.go.erb (hunk 1 above)
Comment (toVersus) [reply mmv1/third_party/terraform/services/container/node_config.go.erb]: Thank you for your assistance! You are right; this change worked perfectly.

**Comment by shuyama1** [comment]
don't think the failing tests are related 
/gcbrun

**Review by shuyama1 — APPROVED** [review]
